foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/EnableHabushuBuildCacheMigration.java

```diff
@@ -0,0 +1,116 @@
+import javax.xml.transform.*;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringReader;
+
+
+/**
+ * This migration adds a new build directory tag to the build node of the project's pom.xml file.
+ */
+public class EnableHabushuBuildCacheMigration extends AbstractAissembleMigration {
+    private static final String BUILD_NODE_PATH = "/project/build";
+    private static final String PACKAGING_NODE_PATH = "/project/packaging";
+    private static final String BUILD_DIR_NODE_PATH = "/project/build/directory";
+
+    // XSL string to strip extra empty newlines from the migrated file-- negates a yucky side effect of the transformer
+    private static final String xsl =
+            "<xsl:stylesheet version=\"1.0\" xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\">\n" +
+            "  <xsl:output indent=\"yes\"/>\n" +
+            "  <xsl:strip-space elements=\"*\"/>\n" +
+            "\n" +
+            "  <xsl:template match=\"@*|node()\">\n" +
+            "    <xsl:copy>\n" +
+            "      <xsl:apply-templates select=\"@*|node()\"/>\n" +
+            "    </xsl:copy>\n" +
+            "  </xsl:template>\n" +
+            "\n" +
+            "</xsl:stylesheet>";
+
+    /**
+     * Retrieves the xml node at the specified path
+     *
+     * @param file The xml file to parse
+     * @param expression The path to the target node
+     * @return The retrieved node, or null if the node at the specified path does not exist.
+     */
+    public static Node getNode(File file, String expression) {
+        DocumentBuilderFactory builderFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder;
+        try {
+            builder = builderFactory.newDocumentBuilder();
+            Document xmlDocument = builder.parse(file);
+            XPath xPath = XPathFactory.newInstance().newXPath();
+            return (Node) xPath.compile(expression).evaluate(xmlDocument, XPathConstants.NODE);
+        } catch (ParserConfigurationException | XPathExpressionException | IOException | SAXException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private boolean isHabushuPackaged(File file) {
+        Node node = getNode(file, PACKAGING_NODE_PATH);
+        return node != null && node.getTextContent().equalsIgnoreCase("habushu");
+    }
+
+    private boolean doesBuildDirTagExist(File file) {
+        Node buildDirNode = getNode(file, BUILD_DIR_NODE_PATH);
+        return buildDirNode != null;
+    }
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        return !doesBuildDirTagExist(file) && isHabushuPackaged(file);
```

Comment:
A: Since this is a POM migration, we can actually just interrogate the maven project object for this info to avoid double-parsing and make the code a little less complex.  May not work super well for build dir since we're checking for explicit specification, but does simplify the packaging check.